.github/workflows/tests.yml

```diff
@@ -0,0 +1,79 @@
+name: Tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  test_linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - name: tests-linux-0
+            shards: 3
+            shard-index: 0
+          - name: tests-linux-1
+            shards: 3
+            shard-index: 1
+          - name: tests-linux-2
+            shards: 3
+            shard-index: 2
```

Comment:
this seems to result in quite ugly shard names like `Tests / test_linux (tests-linux-0, 3, 0) (pull_request)`, any chance we can make it give it pretty names like just `Tests for Linux, shard 0 of 3` or something?